templates/term-meta/partials/input.php

```diff
@@ -2,28 +2,44 @@
 
 declare(strict_types=1);
 
+use Fau\DegreeProgram\Infrastructure\Dashboard\TermMeta\TermMetaFieldValidationPattern;
+
 /**
- * @var array{key: string, value: string, title: string, description: string, type: string} $data
+ * @var array{
+ *      key: string,
+ *      value: string,
+ *      title: string,
+ *      description: string,
+ *      type: string,
+ *      validationPattern: ?TermMetaFieldValidationPattern
+ * } $data
  */
 
 [
     'key' => $key,
     'value' => $value,
     'description' => $description,
     'type' => $type,
+    'validationPattern' => $validationPattern,
 ] = $data;
 
 ?>
 
 
-<input name="<?= esc_attr($key) ?>"
-       id="<?= esc_attr($key) ?>"
-       type="<?= esc_attr($type) ?>"
-       value="<?= esc_attr($value) ?>"
-       size="40"
-       <?php if ($description) : ?>
-           aria-describedby="<?= esc_attr($key) ?>-description"
-       <?php endif; ?>
+<input
+    name="<?= esc_attr($key) ?>"
+    id="<?= esc_attr($key) ?>"
+    type="<?= esc_attr($type) ?>"
+    value="<?= esc_attr($value) ?>"
+    size="40"
+    <?php if ($description) : ?>
+        aria-describedby="<?= esc_attr($key) ?>-description"
+    <?php endif ?>
+
+    <?php if (! is_null($validationPattern)) : ?>
+        pattern="<?= esc_attr($validationPattern->pattern()) ?>"
+        title="<?= esc_attr($validationPattern->expectedPatternMessage()) ?>"
```

Comment:
I wouldn't use the `title` for providing context for the expected value format. This attribute is the last one checked by assistive technology when deciding what to announce to users and generally hidden so chances are high that this info is going to be missed.
Have you considered adding the expected format as part of the field description? That way it's continuously visible to all users, and helps prevent errors even before users input something.

Reply:
<img width="756" alt="Screenshot 2024-05-13 at 10 57 10 AM" src="https://github.com/RRZE-Webteam/FAU-Studium/assets/25235298/66315f6c-a59f-4071-8a6b-534a167f174c">

The `title` attribute is picked up by the browser and displayed on invalid input.
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern#specifying_a_pattern

Reply:
Thanks for the screenshot and the explanation. Since this is not standard behavior though and depends on browser-specific implementations, I would still advice adding the pattern text as part of the field description. Of course feel free to keep the `title` as well so that validation messages appear as they do now where possible.

References from the [accessibility considerations section](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern#usability_and_accessibility_considerations) in the page you linked above:
> provide a description of the pattern in visible text near the control.

> Additionally, assistive technology may read the title aloud when the control gains focus, but this should not be relied upon for accessibility.

> Only relying on the title attribute for the visual display of text content is discouraged, as many user agents do not expose the attribute in an accessible manner.

Reply:
Thanks for the reference, it's nice to know, The description text however already includes information about expected pattern ([example](https://github.com/RRZE-Webteam/FAU-Studium/pull/130/files#diff-edce1703e72164d2986031a6012c9d659a55b9f2462dcbc58bd5e25e3b01ac25R124)) or am I overlooking something?

Reply:
Ahh of course. I overlooked this. Sorry for the noise.